### ISSUE TITLE
bundling failed: NotFoundError: Cannot find entry file index.js in any of the roots

### ISSUE THREAD

**Issue by ryota-murakami** [facebook/react-native#17582]
### Is this a bug report?
No

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

  Environment:
    OS:  macOS High Sierra 10.13.2
    Node:  9.4.0
    Yarn:  1.3.2
    npm:  5.6.0
    Watchman:  4.9.0
    Xcode:  Xcode 9.2 Build version 9C40b

  Packages: (wanted => installed)
    react-native: 0.52.0
    react: 16.2.0

  Target Platform: iOS (11.2)

### Steps to Reproduce

Before, I was `rm -rf ios` and update react-native `0.51.0` -> `0.52.0` and run `react-native upgrade`,
in order to attempt fixing IOS simulator extreamly slow problem.
(finaly, it cause `Simulator -> Debug -> Slow Animations` checked)
<img width="596" alt="screen shot 2018-01-14 at 0 51 44" src="https://user-images.githubusercontent.com/5501268/34907673-8b0040fa-f8c5-11e7-9c8c-6bfe82460aa1.png">

1. yarn start (node node_modules/react-native/local-cli/cli.js start)
2. yarn run-ios (node_modules/.bin/react-native run-ios)
3. facing `bundling failed: NotFoundError: Cannot find entry file index.js in any of the roots`

### Expected Behavior

react-native server detect `index.ios.js` and bundle it.

### Actual Behavior

react-native server  doesn't detect `index.ios.js` and faild bundle it.

<img width="1230" alt="screen shot 2018-01-14 at 0 45 49" src="https://user-images.githubusercontent.com/5501268/34907606-abcbd2c8-f8c4-11e7-9102-79bbd5fc674b.png">

i don't know why react-native server say `index.js` file? without my `index.ios.js`.
whatever, have you any advice anyone? 
thank you so mush:bow:

**Comment by arma7x** [comment]
[Breaking changes and deprecations](https://github.com/facebook/react-native/releases?after=v0.49.3)

**Comment by ryota-murakami** [comment]
@arma7x

It’s my fault about all...
thanks!

**Comment by guyzmo** [comment]
I just ran into the same issue after upgrading my project from <0.49 into >0.50.

I used the `react-native-git-upgrade` tool. It would have been lovely if it could handle the indexes merge automagically, and/or have the `react-native` tool show an error (or warning) to merge both indexes into a single index.

TBH, I might have chased my tail longer if it wasn't for this issue. Thanks @ryota-murakami and @arma7x !
